pycape/cape.py

```diff
@@ -111,6 +112,41 @@ def connect(self, function_ref, function_token=None):
         function_ref = _convert_to_function_ref(function_ref)
         self._loop.run_until_complete(self._connect(function_ref, function_token))
 
+    @contextlib.contextmanager
+    def connection_context(self, function_ref, function_token=None):
+        """Context manager to connect to the enclave hosting the function denoted
+        by `function_ref`.
+
+        Note that this context manager accomplishes the same functionality as
+        the connect method except it will automatically close the connection and
+        reset internal websocket connection's states when exiting the context.
+
+        Args:
+            function_ref: A function ID string or FunctionRef representing a deployed
+                Cape function. If a FunctionRef, can also include the function hash,
+                which  allows the user to verify that the enclave is hosting the same
+                function they deployed.
+            function_token: Optional string containing a Cape function token generated
+                by the Cape CLI during `cape token`. If None, the Cape access token
+                will be used instead.
+
+        Raises:
+            RuntimeError if the websocket response or the enclave attestation doc is
+                malformed, or if the enclave fails to return a function hash matching
+                our own.
+            Exception if the enclave threw an error while trying to fulfill the
+                connection request.
+        """
+        function_ref = _convert_to_function_ref(function_ref)
+        try:
+            yield self._loop.run_until_complete(
+                self._connect(function_ref, function_token)
+            )
```

Comment:
```suggestion
        try:
            yield self.connect(function_ref, function_token)
```

Reply:
we are essentially copying the code from `connect` here, might as well use it to avoid duplication